### ISSUE TITLE
Grace Day mismatch due to auto-created and deleted submission

### ISSUE THREAD

**Issue by TheodorJ** [autolab/Autolab#1031]
Given the following scenario:

1. A course instructor uploads grades for a specific problem in an assessment. For all students who did not submit the assignment, a new submission is made and their grace days used are counted.
2. The instructor deletes the auto-created submissions, and grace days used is not updated for  the students who did not submit the assignment.

A student's count of remaining grace days will be incorrect.

**Comment by victorhuangwq** [comment]
@xinyis991105 sorry for the sudden tag. But this does seem within your ball park, would it be possible for you to check if this issue still exist?

**Comment by xinyis991105** [comment]
I will look into that shortly!

**Comment by xinyis991105** [comment]
I've tested it out on my local master branch and it doesn't seem to consume grace days used when the score is uploaded by the instructor. I made sure that it's not the grace days update scheme that is dysfunctional by submitting from the student's end. In that scenario, the number of grace days used is correctly consumed. The same behavior follows when I tested on the demonstration site. But before closing this issue, I wanted to make sure does uploading scores from an instructor's end refer to bulk import for a particular assessment? Thanks!
